### ISSUE TITLE
Custom BlotName leads to 'The given range isn't in document' for Certain Editor Interactions

### ISSUE THREAD

**Issue by eugenkiss** [quilljs/quill#1133]
When creating a custom blot with a non-default `blotName` (e.g. `link2` instead `link`) certain interactions in the editor lead to the error 'The given range isn't in document'. It's easiest to see the reproduction in the following video: https://streamable.com/8vfd

**Steps for Reproduction**

1. Visit http://codepen.io/quill/pen/oLVKRa?editors=0010 (it's one of the examples of the 'Cloning Medium with Parchment' guide)
2. Change line 28 from `LinkBlot.blotName = 'link';` to `LinkBlot.blotName = 'link2';`, change line 46 from `quill.format('link', value);` to `quill.format('link2', value);` and rerun the codepen.
3. Open the developer console
4. Apply the link format to 'story...' with e.g. 'bla' as the link value in the prompt
5. Put the cursor at the end of 'story...'
6. Press 'Enter', enter a character, delete the character by pressing 'Backspace', observe that you need to do it twice to remove it.
7. The cursor should now be again at the end of 'story...'. Now press 'Enter' and observe that no newline is inserted but instead the developer console reports 'The given range isn't in document'.

**Expected behavior**:

No error. If you do not change the blot name from `link` to `link2` there will be no error (second part of the video).

**Actual behavior**:

Error 'The given range isn't in document'.

**Platforms**:

Tested on macOS Sierra on both Chrome 54 and recent Safari version.

**Version**:

Runnin `Quill.version` in the codepen results in '1.0.0', but I saw the same behavior in my own project with version '1.1.5'.

**Comment by jhchen** [comment]
The Medium guide is to demonstrate how to customize the content of Quill using Parchment and for the purpose of the guide, it uses the bare basics without all keyboard and selection interactions included, so the guide does not produce a fully robust production ready editor--that is what the full Quill editor is for. If you change the import from quill.core.js to quill.js and follow the given instructions, no error occurs.

**Comment by eugenkiss** [comment]
Thanks for the response @jhchen. It seems I'm blind. Can you explain where I would change the import from `quill.core.js` to `quill.js` in the CodePen? 

The settings modal already shows `quill.js` for me:

![screen shot 2016-12-08 at 4 31 39 pm](https://cloud.githubusercontent.com/assets/211365/21015913/f94044a0-bd63-11e6-9da3-9fbbf8555105.png)

**Comment by jhchen** [comment]
Apologies I was not thorough in following the reproduction steps.

First the quill.js vs quill.core.js issue I did not realize that that was the official pen and it was actually being saved so when I was tinkering it was left at quill.js, which is what I tried.

Second browsers have an issue with empty `<a>` tag behavior on newlines so Quill's keyboard module fixes this but whitelists the name "link". It would be prohibitive performance-wise to perform this check dynamically. So renaming to "link2" defeats this fix. This is unique only to blots using the anchor tag so renaming to bold2, italic2, etc should work just fine.
